modules/javafx.graphics/src/test/java/test/javafx/css/Node_cssStyleMap_Test.java

```diff
@@ -73,9 +73,14 @@ public Node_cssStyleMap_Test() {
 
     boolean disabled = false;
 
+    /**
+     * @param property - must be a StyleableProperty
+     * @param map
+     * @param decls
+     */
     private void checkFoundStyle(Property<?> property, Map<StyleableProperty<?>, List<Style>> map, List<Declaration> decls) {
 
-        List<Style> styles = map.get(property);
+        List<Style> styles = map.get((StyleableProperty<?>)property);
```

Comment:
If the `Property` must be a `StyleableProperty` then maybe the method signature should be that and the cast should be the responsibility of the callers.

Reply:
It cannot.  Shape.fillProperty(), for example, is declared as 
public final ObjectProperty<Paint> fillProperty()

ObjectProperty is not a StyleableProperty, though the actual type of the underlying property is.  Same for strokeProperty, strokeWidthProperty etc. 

Also, the existing test code does the typecast later.  

I do agree with Kevin's suggestion to add an assert.